src/protocols/bdx/tests/TestBdxTransferSession.cpp

```diff
@@ -129,8 +129,13 @@ void VerifyStatusReport(nlTestSuite * inSuite, void * inContext, const System::P
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
     NL_TEST_ASSERT(inSuite, payloadHeader.GetProtocolID() == Protocols::kProtocol_Protocol_Common);
     NL_TEST_ASSERT(inSuite, payloadHeader.GetMessageType() == static_cast<uint8_t>(Protocols::Common::MsgType::StatusReport));
+    if (headerSize > msg->DataLength())
+    {
+        NL_TEST_ASSERT(inSuite, false);
+        return;
+    }
 
-    Encoding::LittleEndian::Reader reader(msg->Start() + headerSize, msg->DataLength());
+    Encoding::LittleEndian::Reader reader(msg->Start() + headerSize, static_cast<uint16_t>(msg->DataLength() - headerSize));
```

Comment:
Probably OK if we need to not change the state of `msg`.  Otherwise we could `msg->Consume(headerSize)` or something and then ditch the arithmetic.